ffs/api/api_jobs.go

```diff
@@ -8,6 +8,18 @@ import (
 	"github.com/textileio/powergate/ffs/scheduler"
 )
 
+// GetStorageJob returns the current state of the specified job.
+func (i *API) GetStorageJob(ctx context.Context, jid ffs.JobID) (ffs.StorageJob, error) {
+	job, err := i.sched.GetJob(jid)
+	if err == scheduler.ErrNotFound {
+		return ffs.StorageJob{}, fmt.Errorf("job id %s not found", jid)
```

Comment:
Having this urge to only return machine readable custom error types so that the rpc layer can check what error it is and start returning proper grpc error codes, and do the nice string wrapped error messages at that level. It's tough because I know this file would be considered the public API for a go developer embedding Powergate in their program. But so far, the number of developers we have doing that is 0, and this is making it hard to return proper errors in the RPC layer where we have lots of developers using that API.